### PR TITLE
Add authentication to grpc client calls

### PR DESCRIPTION
**Pull request by klara-l**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ liiklus.target=grpc://liiklus:6565
 
 // or read/write separately, target will be used as a fallback
 liiklus.read.uri=grpc://liiklus:6565
+liiklus.read.secret=SOME_SECRET_FROM_SSM
```
Comment (lanwen) [inline README.md]:
```suggestion
liiklus.read.secret=secret
```

File: starter/src/main/java/io/vivy/liiklus/LiiklusAutoConfiguration.java [hunk 2]
```diff
@@ -53,21 +53,24 @@
     @Autowired
     LiiklusProperties properties;
 
-    private static LiiklusClient createClient(URI targetURI) {
-        if (targetURI == null) {
+    @Autowired
+    Clock clock;
```
Comment (lanwen) [inline starter/src/main/java/io/vivy/liiklus/LiiklusAutoConfiguration.java]: please use arg injection, as you can create bean internally

File: starter/src/main/java/io/vivy/liiklus/auth/grpc/JWTClientInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+            MethodDescriptor<ReqT, RespT> method,
+            CallOptions callOptions,
+            Channel next
+    ) {
+        return new ForwardingClientCall.SimpleForwardingClientCall<>(next.newCall(method, callOptions)) {
+            @Override
+            public void start(Listener<RespT> responseListener, Metadata headers) {
+                log.info("ADDING AUTHENTICATION...");
```
Comment (lanwen) [inline starter/src/main/java/io/vivy/liiklus/auth/grpc/JWTClientInterceptor.java]: not needed

File: starter/src/main/java/io/vivy/liiklus/auth/grpc/JWTClientInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    public static final String AUTHENTICATION_HEADER = "authentication";
+    public static final Duration EXPIRY = Duration.ofMinutes(2);
+
+    Algorithm algorithm;
+
+    Clock clock;
+
+    public JWTClientInterceptor(Clock clock, String secret) {
+        this.clock = clock;
+        algorithm = Algorithm.HMAC512(secret);
```
Comment (lanwen) [inline starter/src/main/java/io/vivy/liiklus/auth/grpc/JWTClientInterceptor.java]: provide algorithm right in the config and remove the constructor override

**Comment by lanwen** [comment]
https://github.com/avast/grpc-java-jwt

**Comment by lanwen** [comment]
aand https://github.com/grpc/grpc-java/blob/master/SECURITY.md#mutual-tls

**Review by belmatter — APPROVED** [review]

**Pull request merged**